### PR TITLE
fix(torchx): proper support for :axes in broadcast

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

**Comment by josevalim** [comment]
This closes #683, right?

File: torchx/test/torchx/nx_doctest_test.exs [hunk 1]
```diff
@@ -15,7 +15,7 @@ defmodule Torchx.NxDoctestTest do
 
   @temporarily_broken_doctests [
     # broadcast - shape mismatch in one test
-    broadcast: 3,
+    # broadcast: 3,
```
Comment (josevalim) [inline torchx/test/torchx/nx_doctest_test.exs:18]:
```suggestion
```

File: torchx/test/torchx/nx_doctest_test.exs [hunk 2]
```diff
@@ -15,7 +15,7 @@ defmodule Torchx.NxDoctestTest do
 
   @temporarily_broken_doctests [
     # broadcast - shape mismatch in one test
```
Comment (josevalim) [inline torchx/test/torchx/nx_doctest_test.exs:17]:
```suggestion
```

**Review by josevalim — APPROVED** [review]

**Comment by polvalente** [comment]
> This closes #683, right?

Yes!

**Pull request merged**